client/state/data-layer/wpcom/me/notification/settings/index.js

```diff
@@ -3,65 +3,56 @@
 /**
  * External dependencies
  */
-
 import { translate } from 'i18n-calypso';
 
 /**
  * Internal dependencies
  */
 import { http } from 'state/data-layer/wpcom-http/actions';
-import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { dispatchRequestEx } from 'state/data-layer/wpcom-http/utils';
 import { NOTIFICATION_SETTINGS_REQUEST } from 'state/action-types';
 import { updateNotificationSettings } from 'state/notification-settings/actions';
 import { errorNotice } from 'state/notices/actions';
 
 /**
- * Dispatches a request to fetch the current user notification settings
+ * Returns an action for HTTP request to fetch the current user notification settings
  *
- * @param   {Function} dispatch Redux dispatcher
  * @param   {Object}   action   Redux action
- * @returns {Object}            dispatched http action
+ * @returns {Object}            http action
  */
-export const requestNotificationSettings = ( { dispatch }, action ) =>
-	dispatch(
-		http(
-			{
-				apiVersion: '1.1',
-				method: 'GET',
-				path: '/me/notifications/settings',
-			},
-			action
-		)
+export const requestNotificationSettings = action =>
+	http(
+		{
+			apiVersion: '1.1',
+			method: 'GET',
+			path: '/me/notifications/settings',
+		},
+		action
 	);
 
 /**
- * Dispatches a notification settings receive action then the request succeeded.
+ * Returns a notification settings receive action then the request succeeded.
  *
- * @param   {Function} dispatch  Redux dispatcher
  * @param   {Object}   action    Redux action
  * @param   {Object}   settings  raw notification settings object returned by the endpoint
- * @returns {Object}             disparched user devices add action
+ * @returns {Object}             notification settings update action
  */
-export const updateSettings = ( { dispatch }, action, settings ) =>
-	dispatch(
-		updateNotificationSettings( {
-			settings,
-		} )
-	);
+export const updateSettings = ( action, settings ) => updateNotificationSettings( settings );
 
 /**
- * Dispatches a error notice action when the request fails
+ * Returns an error notice action when the request fails
  *
- * @param   {Function} dispatch Redux dispatcher
- * @returns {Object}            dispatched error notice action
+ * @returns {Object}   error notice action
  */
-export const handleError = ( { dispatch } ) =>
-	dispatch(
-		errorNotice( translate( "We couldn't load your notification settings, please try again." ) )
-	);
+export const handleError = () =>
+	errorNotice( translate( "We couldn't load your notification settings, please try again." ) );
 
 export default {
 	[ NOTIFICATION_SETTINGS_REQUEST ]: [
-		dispatchRequest( requestNotificationSettings, updateSettings, handleError ),
+		dispatchRequestEx( {
+			fetch: requestNotificationSettings,
+			onSuccess: updateSettings,
+			onError: handleError,
+		} ),
```

Comment:
Maybe a controversial idea, as it makes "test coverage" smaller, but: what about ditching the unit tests for the handlers (they test a static return value anyway) and inline them into the `dispatchRequestEx` param object? That makes the source file half the size.

Reply:
It would be nice to hear some more opinions on this, but IMHO it is more readable when we have those defined separately and not inlined in the `dispatchRequestEx` call. FWIW, there will be more complex handlers, and they won't be very readable when inlined; and with that in mind, we'd probably want to be consistent with the way we define the handlers throughout Calypso.

Reply:
My opinion is to let it be a problem when it becomes a problem and I love getting rid of these kinds of tests - they don't help us that much while at the same time they add more maintenance burden and slow down our tests. If we want tests then we can test behaviors instead of making sure that the source code hasn't been changed. If we want to make sure the source code isn't changed we can add ownership in Github.

In some of these I've been reverting to the more basic `fetch`, `onSuccess`, and `onError` names because the module provides a level of namespacing and makes the `dispatchRequestEx` call pretty terse as @jsnajdr pointed out.